test/tcp_echo_client/tcp_echo_client.cpp

```diff
@@ -20,17 +20,201 @@
 #include "sal-stack-lwip/lwipv4_init.h"
 #include "EthernetInterface.h"
 #include "mbed-drivers/test_env.h"
+#include "sal/socket_api.h"
 
-void app_start(int argc, char *argv[])
+#ifndef TCP_EC_SOCKET_TEST_TIMEOUT
+#define TCP_EC_SOCKET_TEST_TIMEOUT 1.0f
+#endif
+
+
+static struct socket *tcp_ec_client_socket_g;
+static volatile bool tcp_ec_client_event_done_g;
+static volatile bool tcp_ec_client_rx_done_g;
+static volatile bool tcp_ec_client_tx_done_g;
+static volatile struct socket_tx_info tcp_ec_client_tx_info_g;
+static volatile struct socket_event tcp_ec_client_event_g;
+static volatile int tcp_ec_closed;
+static volatile int tcp_ec_connected;
+
+volatile int tcp_ec_timeout_g;
+static void onTimeout() {
+    tcp_ec_timeout_g = 1;
+}
+
+
+/*****************************************************************************
```

Comment:
- agreed doxygen would be better but this would require more work. Is this something we need to do?